logstash-core/src/main/java/org/logstash/instrument/witness/process/ProcessWitness.java

```diff
@@ -0,0 +1,247 @@
+package org.logstash.instrument.witness.process;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.sun.management.UnixOperatingSystemMXBean;
+import org.logstash.instrument.metrics.Metric;
+import org.logstash.instrument.metrics.gauge.NumberGauge;
+import org.logstash.instrument.witness.MetricSerializer;
+import org.logstash.instrument.witness.SerializableWitness;
+import org.logstash.instrument.witness.schedule.ScheduledWitness;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A scheduled witness for process metrics
+ */
+@JsonSerialize(using = ProcessWitness.Serializer.class)
```

Comment:
Much cleaner than the previous situation with this split out into a separate library. :)